float32 en_r1
float32 en_r2
float32 en_r3
float32 en_r4
---

